src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -40,19 +40,20 @@ public sealed partial class JsonSerializerOptions
             return converters;
         }
 
-        private static List<JsonConverter> GetDefaultConverters()
+        private static JsonConverter[] GetDefaultConverters()
         {
             const int NumberOfConverters = 2;
 
-            var converters = new List<JsonConverter>(NumberOfConverters);
-
+            var converters = new JsonConverter[]
+            {
             // Use a list for converters that implement CanConvert().
-            converters.Add(new JsonConverterEnum());
-            converters.Add(new JsonKeyValuePairConverter());
+                new JsonConverterEnum(),
+                new JsonKeyValuePairConverter()
+            };
 
             // We will likely add collection converters here in the future.
 
-            Debug.Assert(NumberOfConverters == converters.Count);
+            Debug.Assert(NumberOfConverters == converters.Length);
```

Comment:
NumberOfConverters can be deleted, along with this assert.  In fact, this whole method can be removed, leaving just:
```C#
private static readonly JsonConverter[] s_defaultFactoryConverters = new JsonConverter[]
{
    new JsonConverterEnum(),
    new JsonKeyValuePairConverter()
};
```

Reply:
Will do, but could you explain the idea of the previous implementation? It was possible to do the same thing using an in-place list initializer.

Reply:
> It was possible to do the same thing using an in-place list initializer.

Not exactly.  I didn't write it, but presumably the author wanted to ensure that the list was being correctly sized, such that it would allocate a backing array of exactly the required size and not need to grow, e.g. ensuring that if someone accidentally added:
```C#
converters.Add(new SomeCoolNewConverter());
```
without increasing NumberOfConverters to 3, the assert would fail.  And there's no way to write that without having a separate method to include the assert.

Reply:
Missed that. Thanks for pointing out!